src/utils/use-promise/index.tsx

```diff
@@ -4,11 +4,11 @@ import { CancellablePromise } from '@/utils/cancellable-promise'
 import { NetworkError } from '@/api/base'
 
 export const usePromise = <T extends any>(
-  promiseCreator: () => Promise<T> | T,
+  promiseCreator: () => Promise<T> | T | false,
   dependencies: any[] = [promiseCreator],
 ) => {
   const emitError = useErrorEmitter()
-  const [val, setVal] = useState<T | undefined | Error | NetworkError>(
+  const [val, setVal] = useState<T | undefined | Error | NetworkError | false>(
```

Comment:
I'm not sure I understand what the `| false` is for here. I don't remember, I might have suggested you to make this change :joy: but I don't think this is the best option here. I think that `promiseCreator` should return `undefined` if it can't get the data.

I think we should undo the changes to this file, which will probably cause errors in other files, but I think that would be a better change.